pkg/storage/ingest/reader.go

```diff
@@ -894,45 +900,57 @@ func (r *concurrentFetchers) pollFetches(ctx context.Context) (result kgo.Fetche
 	}
 }
 
+var errUnknownPartitionLeader = fmt.Errorf("unknown partition leader")
+
+// fetchMinBytesWaitTime is the time the Kafka broker can wait for MinBytes to be filled.
+// This is usually used when there aren't enough records available to fulfil MinBytes, so the broker waits for more records to be produced.
+var fetchMinBytesWaitTime = 10 * time.Second
```

Comment:
nit: I tend to discourage the use of global variables for methods that have a pointer receiver. You could simplify this and add this as an attribute to `concurrentFetchers` then rename this  to `defaultFetchMinBytesWaitTime` which is set on the constructor.

Also, non-nit can you please specify the unit in the name, it was non-obvious that even though this in seconds, the actual client requires millis.

Reply:
Very similar to what you have done with `noReturnedRecords` which btw, I'd suggest you consider moving that to the top too.

Reply:
> Also, non-nit can you please specify the unit in the name, it was non-obvious that even though this in seconds, the actual client requires millis.

it's a `time.Duration`, isn't the type enough? It's effectively in nanoseconds, but it's a proper type

Reply:
the reason i didn't want to do it in the config is because it requires a lot of prod code changes just to ease tests. Take a look at this - a404ec94168f8ee11858037ba42e6b8fb838e3fe - if you like it, we can keep it

Reply:
I was thinking much simpler than the above:

```git
diff --git a/pkg/storage/ingest/reader.go b/pkg/storage/ingest/reader.go
index 854dd0aaf..3bfbc3261 100644
--- a/pkg/storage/ingest/reader.go
+++ b/pkg/storage/ingest/reader.go
@@ -45,6 +45,8 @@ const (
 
 	// kafkaOffsetEnd is a special offset value that means the end of the partition.
 	kafkaOffsetEnd = int64(-1)
+
+	defaultMinBytesWaitTimeSeconds = 10 * time.Second
 )
 
 var (
@@ -840,7 +842,7 @@ func newConcurrentFetchers(ctx context.Context, client *kgo.Client, logger log.L
 		partitionID:        partition,
 		metrics:            metrics,
 		recordsPerFetch:    recordsPerFetch,
-		minBytesWaitTime:   minBytesWaitTime,
+		minBytesWaitTime:   defaultMinBytesWaitTimeSeconds,
 		lastReturnedRecord: noReturnedRecords,
 		tracer:             kotel.NewTracer(kotel.TracerPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}))),
 		orderedFetches:     make(chan kgo.FetchPartition),

```

Reply:
oh, that works too. I prefer that. Done in a6b2e87c59aff7881a2a24ced5e2ee98407739c7